src/Microsoft.Health.Dicom.Tests.Common/Comparers/CustomTagEntryEqualityComparer.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Health.Dicom.Core.Features.CustomTag;
+
+namespace Microsoft.Health.Dicom.Tests.Common.Comparers
+{
+    public class CustomTagEntryEqualityComparer : IEqualityComparer<CustomTagEntry>
+    {
+        public bool Equals(CustomTagEntry x, CustomTagEntry y)
+        {
+            if (x == null || y == null)
+            {
+                return x == y;
+            }
+
+            return x.Path.Equals(y.Path, StringComparison.OrdinalIgnoreCase)
+                && string.Equals(x.VR, y.VR, StringComparison.OrdinalIgnoreCase)
+                && x.PrivateCreator == y.PrivateCreator
+                && x.Level == y.Level
+                && x.Status == y.Status;
+        }
+
+        public int GetHashCode([DisallowNull] CustomTagEntry obj)
```

Comment:
Unless the `nullable` context is enable, this attribute doesn't provide any analysis. Furthermore, it doesn't prevent `null` values from being passed; it is simply to convey what _should_ happen. So you'll need a `null` check still!